views/DonInfo/DonInfo.jsx

```diff
@@ -0,0 +1,231 @@
+import React from 'react';
+import { AsyncStorage, FlatList, TouchableOpacity, Text, View } from 'react-native';
+import { Container, Content, Spinner, Icon } from 'native-base';
+import { Font} from "expo";
+import UserProfile from '../../components/UserProfile/UserProfile.jsx';
+import DonStatusCard from '../../components/DonStatusCard/DonStatusCard';
+import styles from './DonInfoStyle';
+import api from '../../ApiClient';
+
+export default class DonInfo extends React.Component {
+  static navigationOptions = ({ navigation }) => {
+    return {
+      title: 'Find a Don',
+      headerTintColor: '#FFFFFF',
+      headerStyle: {
+        backgroundColor: '#fc4970'
+      },
+      get headerRight() {
+        var state = navigation.getParam('saveState');
+        if (!state || state === 'disabled') return null;
+        else if (state === 'hasChanges') {
+          return (
+            <TouchableOpacity onPress={navigation.getParam('save')}>
+              {
+                <Icon
+                  type="MaterialIcons"
+                  name="save"
+                  style={styles.icon}
+                />
+              }
+            </TouchableOpacity>
+          )
+        } else if (state === 'saving') {
+          return (
+            <View style={{ marginRight: 20 }}>
+              <Spinner
+                size='small'
+                color='white'
+              />
+            </View>
+          )
+        } else if (state === 'saved') {
+          return (
+            <Icon
+              type="MaterialIcons"
+              name="check"
+              style={styles.icon}
+            /> 
+          )
+        }
+      }
+    }
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      dons: [],
+      loading: true,
+      userDataToShow: undefined,
+      showProfileModal: false
+    }
+  }
+
+  async componentWillMount() {
+    await Font.loadAsync({
+      Roboto: require("native-base/Fonts/Roboto.ttf"),
+      Roboto_medium: require("native-base/Fonts/Roboto_medium.ttf")
+    });
+    await this.loadDons();
+  }
+
+  componentDidMount() {
+    this.props.navigation.setParams({
+      save: this.onSave,
+      saveState: 'disabled'
+    });
+  }
+
+  filterDons(users){
+    var dons = []
+    users.forEach(user =>{
+      if((user.role && user.role.includes("don"))==1){
```

Comment:
Not sure if it matters but why the `==1`

Reply:
Because that's the C style guide that I'm used to... will fix :)